drivers/usb/device/usb_dc_stm32.c

```diff
@@ -959,11 +963,15 @@ int usb_dc_ep_mps(const uint8_t ep)
 
 int usb_dc_detach(void)
 {
-	LOG_ERR("Not implemented");
 	if (!usb_dc_stm32_state.attached) {
 		return 0;
 	}
 
+	irq_disable(USB_IRQ);
+
+	/* Enable soft disconnect */
+	USB_DEVICE_REG->DCTL |= USB_OTG_DCTL_SDIS;
+
```

Comment:
Why not using USB_DevDisconnect() ? which seems HAL equivalent

Reply:
See previous comment.